app/src/test/java/org/simple/clinic/bp/entry/BloodPressureValidationMockDateValidatorTest.kt

```diff
@@ -0,0 +1,168 @@
+import org.simple.clinic.util.RxErrorsRule
+import org.simple.clinic.util.TestUserClock
+import org.simple.clinic.util.TestUtcClock
+import org.simple.clinic.util.UserInputDatePaddingCharacter
+import org.simple.clinic.widgets.UiEvent
+import org.simple.clinic.widgets.ageanddateofbirth.UserInputDateValidator
+import org.simple.clinic.widgets.ageanddateofbirth.UserInputDateValidator.Result
+import org.simple.clinic.widgets.ageanddateofbirth.UserInputDateValidator.Result.Invalid.DateIsInFuture
+import org.simple.clinic.widgets.ageanddateofbirth.UserInputDateValidator.Result.Invalid.InvalidPattern
+import org.threeten.bp.LocalDate
+import java.util.UUID
+
+@RunWith(JUnitParamsRunner::class)
+class BloodPressureValidationMockDateValidatorTest {
+
+  @get:Rule
+  val rxErrorsRule = RxErrorsRule()
+
+  private val ui = mock<BloodPressureEntryUi>()
+  private val bloodPressureRepository = mock<BloodPressureRepository>()
+  private val appointmentRepository = mock<AppointmentRepository>()
+  private val patientRepository = mock<PatientRepository>()
+  private val dateValidator = mock<UserInputDateValidator>()
+  private val bpValidator = BpValidator()
+
+  private val uiEvents = PublishSubject.create<UiEvent>()
+  private val patientUuid = UUID.fromString("79145baf-7a5c-4442-ab30-2da564a32944")
+
+  private val testUserClock = TestUserClock()
+  private val testUtcClock = TestUtcClock()
+  private val userSession = mock<UserSession>()
+
+  private val facilityRepository = mock<FacilityRepository>()
+  private val user = PatientMocker.loggedInUser(uuid = UUID.fromString("1367a583-12b1-48c6-ae9d-fb34f9aac449"))
+
+  private val facility = PatientMocker.facility(uuid = UUID.fromString("2a70f82e-92c6-4fce-b60e-6f083a8e725b"))
+  private val userSubject = PublishSubject.create<User>()
+
+  private val controller = BloodPressureEntrySheetController(
+      bloodPressureRepository = bloodPressureRepository,
+      appointmentRepository = appointmentRepository,
+      patientRepository = patientRepository,
+      dateValidator = dateValidator,
+      bpValidator = bpValidator,
+      userClock = testUserClock,
+      inputDatePaddingCharacter = UserInputDatePaddingCharacter('0'),
+      userSession = userSession,
+      facilityRepository = facilityRepository)
+
+  @Before
+  fun setUp() {
+    RxJavaPlugins.setIoSchedulerHandler { Schedulers.trampoline() }
+
+    whenever(dateValidator.dateInUserTimeZone()).doReturn(LocalDate.now(testUtcClock))
+    whenever(userSession.requireLoggedInUser()).doReturn(userSubject)
+    whenever(facilityRepository.currentFacility(user)).doReturn(Observable.just(facility))
+
+    uiEvents
+        .compose(controller)
+        .subscribe { uiChange -> uiChange(ui) }
+
+    userSubject.onNext(user)
+  }
+
+  // TODO(rj): 2019-10-10 There isn't a straight-forward way to set a time in the date validator as of now. We'll rely on mocks to test this code path (showDateIsInFutureError) for now.
```

Comment:
Do we have a pivotal story for changing the `DateValidator` to accept a `UserClock`?

Reply:
Yes, we do - here is the story.

https://www.pivotaltracker.com/story/show/168775283